### PR TITLE
Add environment variable goproxy

### PR DESCRIPTION
**Pull request by Fkbqf**
### Ⅰ. Describe what this PR did

Add environment variable goproxy  for wasm-go/Dockerfile

### CODE REVIEW

File: plugins/wasm-go/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ARG BUILDER=higress-registry.cn-hangzhou.cr.aliyuncs.com/plugins/wasm-go-builder:go1.19-tinygo0.27.0-oras1.0.0
 FROM $BUILDER as builder
 
+ENV GOPROXY https://goproxy.cn
```
Comment (CH3CHO) [inline plugins/wasm-go/Dockerfile:4]: This may not be a good idea for developers outside the Chinese mainland.

Comment (CH3CHO) [reply plugins/wasm-go/Dockerfile:4]: Check out this: https://github.com/alibaba/higress/blob/main/plugins/wasm-go/Makefile

Allowing executors to configure their own value might be better.

Comment (Fkbqf) [reply plugins/wasm-go/Dockerfile:4]: > Check out this: https://github.com/alibaba/higress/blob/main/plugins/wasm-go/Makefile
> 
> Allowing executors to configure their own value might be better.

But for domestic developers, if you don’t add this option, it’s not very friendly, I didn’t think carefully, you’re right

**Comment by johnlanni** [comment]
Maybe we should remove `go mod tidy` in Dockerfile, then GOPROXY all depends on local workspace.

**Comment by CH3CHO** [comment]
> Maybe we should remove `go mod tidy` in Dockerfile, then GOPROXY all depends on local workspace.

Agree. And we can execute `go mod tidy` locally instead of inside the builder container.

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**